tests/algorithms/test_lbfgs.py

```diff
@@ -0,0 +1,74 @@
+"""Tests for LBFGS."""
+
+# Copyright 2023 Physikalisch-Technische Bundesanstalt
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#       http://www.apache.org/licenses/LICENSE-2.0
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+import pytest
+import torch
+import torch.nn.functional as F
+
+from mrpro.algorithms import lbfgs
+from mrpro.data import SpatialDimension
+from mrpro.operators import ConstraintsOp
+from mrpro.operators import Operator
+from mrpro.operators.functionals import L2_data_discrepancy
+from mrpro.operators.models import SaturationRecovery
+from mrpro.phantoms import EllipsePhantom
+from tests import RandomGenerator
+from tests.operators._OptimizationTestFunctions import Rosenbrock
+
+
+@pytest.mark.parametrize(
+    'a, b',
+    [(1.0, 100.0), (1.0, 50.0), (1.0, 25.0), (1.0, 2.0)],
+)
+def test_lbfgs_rosenbrock(a, b):
+    """Test lbfgs functionality."""
+
+    # with pytest.raises(ImportWarning):  # ToDo: remove this when fixed
```

Comment:
In test_adam this is still needed? Pls remove any commented code before merging

Reply:
I am not sure... when run locally (on e81330), I seem to need it, after having rebased to main. Before it seemed not necessary.

Reply:
it depends on the pytorch version in the environment.
if you don't `pip install` again (or update the environment) you will continue to use the old pytorch version which is older then our requirements in the pyproject.toml